src/service/context/mod.rs

```diff
@@ -151,6 +151,55 @@ impl<S> Context<S> {
     pub fn get<T: Send + Sync + 'static>(&self) -> Option<&T> {
         self.extensions.get::<T>()
     }
+    /// Get a reference to an extension or T's default().
+    ///
+    /// An extension is a type that implements `Send + Sync + 'static`,
```

Comment:
[pedantic] I do not think there is much value in repeating the same documentation everywhere, also harder to maintain. So as an extended doc (everything after the first line) I would simply refer to `Context::get` for more information.

Also might be useful to link the `default()` part above to the actual `Default` std trait.